llvm/lib/Transforms/Utils/CodeExtractor.cpp

```diff
@@ -1585,8 +1585,27 @@ static void fixupDebugInfoPostExtraction(Function &OldFunc, Function &NewFunc,
     return cast<DILocalVariable>(NewVar);
   };
 
-  auto UpdateDPValuesOnInst = [&](Instruction &I) -> void {
-    for (DPValue &DPV : DPValue::filter(I.getDbgValueRange())) {
+  auto UpdateDbgRecordsOnInst = [&](Instruction &I) -> void {
+    for (DbgRecord &DR : I.getDbgValueRange()) {
+      if (DPLabel *DPL = dyn_cast<DPLabel>(&DR)) {
```

Comment:
Would this work with an auto lambda to share the logic for DbgLabelInsts?

Reply:
Possibly - the DPValue code here hasn't been auto-lambda'd. I'll have a play with it.

Reply:
Yeah that looks better, done (added a `setLabel` method to `DbgLabelInst` (intrinsic version)).